src/components/media/video/video.tsx

```diff
@@ -0,0 +1,329 @@
+/* eslint-disable no-magic-numbers */
+import './video.css'
+
+import { useMediaQuery, useTheme } from '@mui/material'
+import Alert from '@mui/material/Alert'
+import CircularProgress from '@mui/material/CircularProgress'
+import Fade from '@mui/material/Fade'
+import Typography from '@mui/material/Typography'
+import { Box } from '@mui/system'
+import React, { useEffect, useRef, useState } from 'react'
+
+import type { VideoFormat } from '../../types'
+import {
+  MoveTenSecondsButton,
+  PlaybackRateButton,
+  PlayOrPauseButton,
+  ProgressSlider,
+  ToggleTranscriptButton,
+  VolumeSettings,
+} from '../controls/commonControls'
+import { MediaIconButton } from '../controls/mediaIconButton'
+import {
+  FullscreenButton,
+  PictureInPictureButton,
+} from '../controls/videoControls'
+import TimeIndicator from '../timeIndicator/timeIndicator'
+import Transcript from '../transcript/transcript'
+
+export default function Video(props: VideoFormat) {
+  const [isTranscriptShown, setIsTranscriptShown] = useState(false)
+  const [areCaptionsShown, setAreCaptionsShown] = useState(false)
+  const [isLoading, setIsLoading] = useState(true)
+  const [isFullscreen, setIsFullscreen] = useState(!!document.fullscreenElement)
+
+  const [controlErrorMessage, setControlErrorMessage] = useState('')
+  const [errorMessage, setErrorMessage] = useState('')
+  const [elapsedTime, setElapsedTime] = useState(0)
+
+  const videoContainerRef = useRef<HTMLElement>(null)
+  const videoRef = useRef<HTMLVideoElement>(null)
+  const theme = useTheme()
+  const isMobile = useMediaQuery(theme.breakpoints.down('md'))
+
+  useEffect(() => {
+    const loadingErrorMessage = 'The video resource has failed to load'
+    const stalledErrorMessage = 'Failed to fetch data, but trying'
+
+    function updateElapsedTime() {
+      if (videoRef.current) {
+        setElapsedTime(videoRef.current.currentTime)
+      }
+    }
+    function handleCanPlay() {
+      setErrorMessage('')
+      setIsLoading(false)
+    }
+    function handleError(errorMessage: string) {
+      setErrorMessage(errorMessage)
+      setIsLoading(false)
+    }
+
+    videoRef.current?.addEventListener('timeupdate', updateElapsedTime)
+    videoRef.current?.addEventListener('canplay', handleCanPlay)
+    videoRef.current?.addEventListener('error', () =>
+      handleError(loadingErrorMessage)
+    )
+    videoRef.current?.addEventListener('stalled', () =>
+      handleError(stalledErrorMessage)
+    )
+    document.addEventListener('fullscreenchange', () => {
+      setIsFullscreen(!!document.fullscreenElement)
+    })
+  }, [])
+
+  function adjustHexColorOpacity(color: string, opacity: number) {
+    const r = parseInt(color.slice(1, 3), 16)
+    const g = parseInt(color.slice(3, 5), 16)
+    const b = parseInt(color.slice(5, 7), 16)
+
+    return `rgba(${r}, ${g}, ${b}, ${opacity})`
+  }
+
+  const backgroundColor = adjustHexColorOpacity(
+    theme.palette.background.paper,
+    0.6
+  )
```

Comment:
I think we should not set a opacity of 0.6 but leave it at 1. In case of videos, it can effect visibility since the video may be running and could have conflicting colors.